usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -314,16 +314,27 @@ def _fetch_parent_award_details(parent_award_ids: dict) -> Optional[OrderedDict]
         logging.debug("Unable to find award for award id %s" % parent_award_award_id)
         return None
 
-    parent_agency = (
+    parent_sub_agency = (
         SubtierAgency.objects.filter(subtier_code=parent_award["latest_transaction__contract_data__agency_id"])
-        .values("name")
+        .values("name", "subtier_agency_id")
         .first()
     )
+    parent_agency = (
+        (
+            Agency.objects.filter(subtier_agency_id=parent_sub_agency["subtier_agency_id"])
```

Comment:
`toptier_flag=True`?

Reply:
I don't think so, the `toptier_flag` is only for the subtier agency that is considered the toptier. So if I was looking for the parent agency of "Federal Acquisition Service" (subtier_agency_id = 634), it's listed in the agency table as having `toptier_agency_id = 40`, which is GSA. The agency table doesn't have any entry with both of these and the `toptier_flag`, because Federal Acquisition Service isn't a toptier agency. The only agency with `toptier_agency_id = 40` is `subtier_agency_id=614`, which is the subtier agency version of GSA.